L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp1.cc

```diff
@@ -0,0 +1,1100 @@
+}
+
+void
+HGCalTriggerGeometryV9Imp1::
+initialize(const edm::ESHandle<CaloGeometry>& calo_geometry)
+{
+    throw cms::Exception("BadGeometry")
+        << "HGCalTriggerGeometryV9Imp1 geometry cannot be initialized with the V7/V8 HGCAL geometry";
+}
+
+void
+HGCalTriggerGeometryV9Imp1::
+initialize(const edm::ESHandle<HGCalGeometry>& hgc_ee_geometry,
+        const edm::ESHandle<HGCalGeometry>& hgc_hsi_geometry,
+        const edm::ESHandle<HGCalGeometry>& hgc_hsc_geometry
+        )
+{
+    // setCaloGeometry(calo_geometry);
+    setEEGeometry(hgc_ee_geometry);
+    setHSiGeometry(hgc_hsi_geometry);
+    setHScGeometry(hgc_hsc_geometry);
+    heOffset_ = eeTopology().dddConstants().layers(true);
+    totalLayers_ = heOffset_ + hsiTopology().dddConstants().layers(true);
+    trigger_layers_.resize(totalLayers_+1);
+    unsigned trigger_layer = 0;
+    for(unsigned layer=0; layer<totalLayers_; layer++)
+    {
+        if(disconnected_layers_.find(layer)==disconnected_layers_.end())
+        {
+            // Increase trigger layer number if the layer is not disconnected
+            trigger_layers_[layer] = trigger_layer;
+            trigger_layer++;
+        }
+        else
+        {
+            trigger_layers_[layer] = 0;
+        }
+    }
+    fillMaps();
+    fillNeighborMapSilicon(l1tCellNeighborsMapping_, trigger_cell_neighbors_);
+    fillNeighborMapScintillator(l1tCellNeighborsSciMapping_, trigger_cell_neighbors_sci_);
+    fillInvalidTriggerCells();
+
+}
+
+unsigned 
+HGCalTriggerGeometryV9Imp1::
+getTriggerCellFromCell( const unsigned cell_id ) const
+{
+    unsigned det = DetId(cell_id).det();
+    unsigned subdet = 0;
+    int zside = 0;
+    unsigned tc_type = 1;
+    unsigned layer = 0;
+    unsigned wafer_trigger_cell = 0;
+    unsigned trigger_cell = 0;
+    // Scintillator
+    if(det == DetId::HGCalHSc)
+    {
+        HGCScintillatorDetId cell_det_id(cell_id);
+        unsigned ieta = cell_det_id.ietaAbs();
+        unsigned iphi = cell_det_id.iphi();
+        tc_type = cell_det_id.type();
+        layer = cell_det_id.layer();
+        subdet = ForwardSubdetector::HGCHEB;
+        zside = cell_det_id.zside();
+        auto trigger_cell_itr = cells_to_trigger_cells_sci_.find(packIetaIphi(ieta, iphi));
+        if(trigger_cell_itr==cells_to_trigger_cells_sci_.end())
+        {
+            throw cms::Exception("BadGeometry")
+                << "HGCalTriggerGeometry: scintillator cell ieta=" << ieta << ", iphi="<<iphi<<" is not mapped to any trigger cell. The trigger cell mapping should be modified.\n";
+        }
+        trigger_cell = 0;
+        wafer_trigger_cell = 0;
+        unpackWaferCellId(trigger_cell_itr->second, wafer_trigger_cell, trigger_cell);
+    }
+    // Silicon
+    else if(det == DetId::HGCalEE || det == DetId::HGCalHSi)
+    {
+        HGCSiliconDetId cell_det_id(cell_id);
+        subdet = (det==DetId::HGCalEE ? ForwardSubdetector::HGCEE : ForwardSubdetector::HGCHEF);
```

Comment:
I would put some comments in the code indicating that the use of the "old" HGCal DetIds for the trigger cells is temporary.